Week1 [Step3] juho

JuhoJUNG-Kr opened the pull request:
@Kioding 
week 1 step3 주문에 따른 빼빼로 만들기 과제입니다. 
매개변수와 상수 그리고 메서드를 호출하는 함수를 만들어 사용자의 주문에 따른 빼빼로를 간편하게 만들 수 있도록 설계했습니다.

JuhoJUNG-Kr committed:
Feat: making ppepero code week1 step3
1.code for making ppepero
2.document for code

-

Kioding commented on OrderPperoStep3/main.swift:
```diff
@@ -0,0 +1,87 @@
+막대길이: 4
+""" )
+
+orderPpero(stickLonger: 4, bodyLonger: 10, body: "***", topping: " ")
```
각 빼빼로 타입에 따른 매개변수를 아래에서 받고 있는데 위에서 해당하는 빼빼로마다 하드코딩을 하고 있네요 👀
정보를 print 하는 부분도 함수처럼 만들어서 재사용을 해볼 수 있을 것 같아요!

JuhoJUNG-Kr replied:
@Kioding 
print 문을 메서드를 호출하는 함수 내부에 넣어서 출력하도록 하는데에는 성공했습니다.
그런데 말씀하신 하드코딩이라는 부분에는 잘 이해가 가지 않는데요... 
함수 안에 매개변수를 받아서 각각의 정보에 따른 빼빼로를 출력하는 함수가 4개인 것보다 더 줄일 수 있다는 말씀이신가요?

Kioding commented on OrderPperoStep3/main.swift:
```diff
@@ -0,0 +1,87 @@
+    for _ in 1...sticklonger{
+        print(" | | ")
+    }
+}
```
- `매개변수 : Type` 의 컨벤션은 어떻게 작성하는 게 좋을까요?!
보통은 `매개변수: Type` 이렇게 통일하여 작성하는 편입니다.
- 중괄호를 사용할 때도 한칸을 띄어쓰면 가독성이 좋아집니다.
- [API Design Guidelines-  Conventions ](https://www.swift.org/documentation/api-design-guidelines/#general-conventions)
링크에서 어떻게 컨벤션을 적용하는지 한번 확인하고 고쳐보면 좋을 것 같아요 😆
위 코멘트는 모든 코드에 대한 코멘트입니다 :)

JuhoJUNG-Kr replied:
변경완료! 주의하겠습니다. :)

Kioding commented on OrderPperoStep3/main.swift:
```diff
@@ -0,0 +1,87 @@
+                print(" |0|")
+            }else{
+                print(" | |")
+            }
```
- `i`가 무슨뜻으로 쓰였을까요?!
처음 코드를 본다면 이해하기 어려울 것 같아요!
swift에서는 축약어를 지양하고 있는데 어떻게 쓰면 좋을 지 고민해볼까요!
- 연산자를 사용할 때 컨벤션은 어떻게 할 수 있을까요? 
아래를 참고하여 이 부분도 그에 맞게 바꿔보면 좋을 것 같습니다 😁
[the swift programming language - Basic Operators](https://docs.swift.org/swift-book/LanguageGuide/BasicOperators.html)

JuhoJUNG-Kr replied:
i를 count 로 변경했습니다!

Kioding commented on OrderPperoStep3/main.swift:
```diff
@@ -0,0 +1,87 @@
+
+func makeStickOfPpero(sticklonger : Int){
+    for _ in 1...sticklonger{
+        print(" | | ")
```
` | | `가 뜻하는 바가 무엇일까요? 반복적으로 사용한다면 상수로 사용하는 건 어떨까요?!

JuhoJUNG-Kr replied:
변경완료 :)

Kioding commented on OrderPperoStep3/main.swift:
```diff
@@ -0,0 +1,87 @@
+    }
+}
+func orderPpero(stickLonger: Int, bodyLonger: Int, body: String, topping: String){
+    makeBodyOfPpero(body: body, topping: topping)
```
`longerOfPpero()` 내부에서 `makeBodyOfPpero()` 를 호출하고 있는데 이 곳에서도 호출하는 이유가 있는 지 궁금합니다 👀

JuhoJUNG-Kr replied:
불필요한 함수였네요! 삭제 했습니다.

Kioding commented on OrderPperoStep3/main.swift:
```diff
@@ -0,0 +1,87 @@
+    print(topping + body + topping)
+}
+
+func longerOfPpero(bodyLonger: Int, body: String, topping: String){
```
함수의 네이밍은 `동사 + 명사` 로 주로 작성하며
실제로 함수의 역할에 대한 네이밍을 지어주면 보다 직관적이랍니다!

주호님께서 다른 메서드에 지어주신 것처럼 이름을 바꿔보면 좋을 것 같네요!

JuhoJUNG-Kr replied:
변경완료!

JuhoJUNG-Kr committed:
Refactor: delete folder for code
1.step2 폴더 삭제
-아이스크림 출력 코드와 빼빼로 출력코드를 한 파일로 합쳐 불필요한 부분을 수정했습니다.

-

JuhoJUNG-Kr committed:
Refactor: change code and document
1.
코드 네이밍 변경
주문내용print문 함수로 수정
순서도 수정
2.
피드백 주신대로 컨벤션과 함수이름을 좀더 직관적인 이름으로 변경했습니다.
순서도에서 문법적 요소를 제거했습니다

-

JuhoJUNG-Kr committed:
Docs:fix document
순서도 수정

-

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -9,22 +9,78 @@
+몸통:\(bodyForm)
+토핑:\(topping)
+막대길이:\(stickHeigth)
+""")
```
내부로 넣어주시면서 매개변수를 사용하여 편리하게 바뀌었네요!
이제 하드코딩을 하지 않아도 되겠어요 😆
정보를 `print` 하는 부분을 함수로 따로 빼내어 `orderPpero()` 에서 호출하는 형식으로 한다면
여기에서는 순서대로 함수를 호출해주는 구나 라고 흐름을 잘 보여줄 수 있을 것 같네요!

<br>

Ppero는 오타일까요? ㅎㅎ

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -9,22 +9,78 @@
+        for count in 1...bodyHeigth {
+            if count%2 != 0 {
+                print(" |0|")
+            }else {
```
else 앞에 띄어쓰기를 해주면 컨벤션이 통일 될 것 같네요 :)

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -9,22 +9,78 @@
-makeBodyOfIcecream(bodyWidth: 10, bodyLonger: 7)
-makeTailOfIcecream(tailLonger: 3)
+makeBodyOfIcecream(icecreamBodyWidth: 10, icecreamBodyLonger: 7)
+makeTailOfIcecream(icecreamTailLonger: 3)
```
step2 에서 했던 코드가 `main.swift` 에 그대로 있네요?!
`main.swift` 는 Code의 Entry Point 로 모든 코드를 읽어옵니다.
하지만, 실제로 사용하지 않는코드라면 어떻게 관리하는 게 좋을까요?!

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -9,22 +9,78 @@
+
+
+
+let tailOfPpero = " | | "
```
상수로 따로 빼주어 어떤 게 들어가는 지 명확히 해주셨네요 👍

JuhoJUNG-Kr committed:
Refactor: change code for step3
1.
-print 구문 함수로 빼내어 출력
-오타수정
-컨벤션 수정
-step2 주석처리

-

JuhoJUNG-Kr committed:
Docs: fix document for code
1.
순서도 수정

-

JuhoJUNG-Kr committed:
Docs: add screen shot

-

JuhoJUNG-Kr committed:
Style: code style change
1.print문의 함수명 수정

-

Kioding commented on CodeStarterCamp_Week1/main.swift:
```diff
@@ -9,22 +9,83 @@
+
+
+func orderPpepero(stickHeigth: Int, bodyHeigth: Int, bodyForm: String, topping: String) {
+    func informationOrder() {
+        print("""
+    길이:\(bodyHeigth)
+    몸통:\(bodyForm)
+    토핑:\(topping)
+    막대길이:\(stickHeigth)
+    """)
+    }
```
함수를 선언했다면 외부에서 선언하여 
`informationOrder(매개변수)` 의 1줄의 형태로 사용하는 게 가독성이 클 것 같네요ㅎㅎㅎ
